backend/Watchdog.Core/Watchdog.Core.BLL/Services/ApplicationService.cs

```diff
@@ -138,5 +138,19 @@ public async Task DeleteApplicationAsync(int appId)
             await _context.SaveChangesAsync();
         }
 
+        public async Task<bool> IsProjectNameValid(string projectName, int organizationId)
+        {
+            if (projectName.Length < 3 || projectName.Length > 50)
+            {
+                return false;
+            }
+
+            return !(await _context.Applications
+                .Include(a => a.Platform)
+                .Where(a => a.OrganizationId == organizationId)
+                .ToListAsync())
```

Comment:
You don't need to use ToListAsync here.
It is a bad decision to load all Applications

Reply:
Fix